### ISSUE TITLE
media-query-no-invalid stylelint issue on using variable inside media query 

### ISSUE THREAD

**Issue by OmPr366** [stylelint/stylelint#7163]
### What minimal example or steps are needed to reproduce the bug?

```css
@mixin minimal-desktop {
  @media (min-width: $lg-breakpoint) {
    @content;
  }
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
"rules": {
        "no-empty-source": true,
        "no-invalid-double-slash-comments": true,
        "no-invalid-position-at-import-rule": true,
        "function-no-unknown": null,
        "selector-class-pattern": [
            "^[a-zA-Z][-_a-zA-Z0-9]*$",
            {
                "message": "Expected class selector alpha numerical with - and _"
            }
        ],
        "selector-max-compound-selectors": 8,
        "max-nesting-depth": [
            7,
            {
                "ignoreAtRules": ["each", "media", "supports", "include"]
            }
        ],
        "selector-max-id": 1,
        "selector-pseudo-element-no-unknown": [
            true,
            {
                "ignorePseudoElements": ["ng-deep"]
            }
        ],
        "no-duplicate-selectors": null,
        "no-descending-specificity": null,
        "annotation-no-unknown": [
            true,
            {
                "ignoreAnnotations": ["default"]
            }
        ],
        "selector-no-qualifying-type": null,
        "scss/at-extend-no-missing-placeholder": null,
        "color-format/format": {
            "format": "rgba"
        },
        "plugin/no-low-performance-animation-properties": null,
        "prettier/prettier": true,
        "declaration-block-no-duplicate-properties": true,
        "import-notation": "string",
        "at-rule-empty-line-before": null,
        "block-opening-brace-space-before": null,
        "color-hex-case": null,
        "declaration-bang-space-after": null,
        "declaration-bang-space-before": null,
        "declaration-block-semicolon-newline-after": null,
        "declaration-block-semicolon-space-before": null,
        "declaration-block-trailing-semicolon": null,
        "declaration-colon-space-after": null,
        "declaration-colon-space-before": null,
        "function-comma-space-after": null,
        "function-parentheses-space-inside": null,
        "media-feature-parentheses-space-inside": null,
        "no-missing-end-of-source-newline": null,
        "number-leading-zero": null,
        "number-no-trailing-zeros": null,
        "selector-list-comma-newline-after": null,
        "string-quotes": null,
        "indentation": null
    }
}
```
### How did you run Stylelint?

```shell
stylelint \"src/**/*.{css,sass,scss,less}\"
```

### Which Stylelint-related dependencies are you using?

```json
{
"stylelint": "^15.10.3",
    "stylelint-color-format": "^1.1.0",
    "stylelint-config-sass-guidelines": "^10.0.0",
    "stylelint-config-standard": "^34.0.0",
    "stylelint-high-performance-animation": "^1.9.0",
    "stylelint-prettier": "^3.0.0",
}
```

### What did you expect to happen?

No error should produce

### What actually happened?

The following problems were reported:

```
src/assets/sass/framework/base/_mixins.scss
 95:10  ✖  Unexpected invalid media query "(min-width:   media-query-no-invalid
           $lg-breakpoint)"

1 problem (1 error, 0 warnings)

```

**Comment by ybiquitous** [comment]
@OmPr366 Thanks for the report with the details. I'm afraid that the `media-query-no-invalid` rule does not support non-standard syntax like SCSS. See the doc below: ⬇️ 

https://github.com/stylelint/stylelint/blob/b509d18ff155d07a3c839639a15040654e47d9d7/lib/rules/media-query-no-invalid/README.md?plain=1#L14

So, I recommend turning off the rule.

**Comment by OmPr366** [comment]
It will be better to add this `media-query-no-invalid` rule for CSS-like languages, such as Sass or Less  or should turn off automatic.

**Comment by ybiquitous** [comment]
`media-query-no-invalid` is turned on by `stylelint-config-recommended` (which is extended by `stylelint-config-standard):
https://github.com/stylelint/stylelint-config-recommended/blob/13.0.0/index.js#L27

So, can you just turn off the rule in your config? Or, you can request a change to the sharable configs you use to turn off it.

Stylelint core rules and official configs focus on standard syntax because we have very little manpower to maintain this project.

**Comment by jeddy3** [comment]
As [per our docs](https://stylelint.io/user-guide/get-started/#linting-css-like-languages-and-css-within-containers) you'll want to use a community config like [stylelint-config-standard-scss](https://www.npmjs.com/package/stylelint-config-standard-scss) if you're linting a CSS-like language.
